OpenSim/Tools/Test/testVisualization.cpp

```diff
@@ -225,23 +241,39 @@ bool testVisModelAgainstStandard(Model& model, const SimTK::Array_<DecorativeGeo
     SimTK::State& si = model.initSystem();
     if (visualDebug)
         model.getVisualizer().show(si);
-    ModelDisplayHints mdh;
+    const ModelDisplayHints& mdh = model.getDisplayHints();
     SimTK::Array_<SimTK::DecorativeGeometry> geometryToDisplay;
     model.generateDecorations(true, mdh, si, geometryToDisplay);
     cout << geometryToDisplay.size() << endl;
     model.generateDecorations(false, mdh, si, geometryToDisplay);
     cout << geometryToDisplay.size() << endl;
-
+    /*
+    DecorativeGeometryImplementationText textFromModel;
+    textFromModel.setPrintTransforms(true); // Debugging
+    for (const SimTK::DecorativeGeometry* nextGeom = geometryToDisplay.begin();
+    nextGeom != geometryToDisplay.end();
+        ++nextGeom) {
+        nextGeom->implementGeometry(textFromModel);
+    }
+    std::cout << textFromModel.getAsString() << std::endl;
+    */
     int i = 0;
     for (const SimTK::DecorativeGeometry* nextGeom = stdPrimitives.begin();
         nextGeom != stdPrimitives.end();
         ++nextGeom) {
             DecorativeGeometryImplementationText dgiTextFromStandard;
+            dgiTextFromStandard.setPrintTransforms(false);
             nextGeom->implementGeometry(dgiTextFromStandard);
             DecorativeGeometryImplementationText dgiTextFromModel;
+            dgiTextFromModel.setPrintTransforms(false);
             geometryToDisplay[i].implementGeometry(dgiTextFromModel);
             if (!(dgiTextFromStandard.getAsString() == dgiTextFromModel.getAsString()))
                 throw  OpenSim::Exception("failed comparing " + dgiTextFromStandard.getAsString() + "vs." + dgiTextFromModel.getAsString());
+            // Compare transforms, this has to be more lenient than String comparison due to roundoff
+            SimTK::Mat44 diffTransform = nextGeom->getTransform().toMat44() - geometryToDisplay[i].getTransform().toMat44();
+            double norm = diffTransform.norm();
+            if (norm > SimTK::Eps) 
+                throw  OpenSim::Exception("Difference in transforms of DecoratibeGeometry:" + std::to_string(norm));
```

Comment:
You should be able to use `SimTK_TEST_EQ` or `SimTK_TEST_EQ_TOL`, which will do an "almost-equal" check.